### ISSUE TITLE
Provide extension for Dbdatasource 

### ISSUE THREAD

**Issue by aygalinc** [DapperLib/Dapper#1885]
Hi,

Is there any plan to support DbDataSource directly with dapper ?

**Comment by goenning** [comment]
What a coincidence! I just learned about DbDataSource today and immediately tried to use with Dapper.

**Comment by 0xced** [comment]
Discussion on the .NET runtime repository for reference: [Introduce DbDataSource abstraction to System.Data](https://github.com/dotnet/runtime/issues/64812)

**Comment by mgravell** [comment]
Yeah, we should get this done. I'll want to think very carefully about which APIs we included though - the existing ones have grown organically rather than as a defined plan. Thanks for reminder - tagging.

**Comment by aygalinc** [comment]
@mgravell do you think a way could be to first create the interface in DapperAot and basically throw NotImplementedException and start to build the code generation for this case ? 
I can give a shot if the api surface is basically the same as the one on dbConnection.

**Comment by mgravell** [comment]
@aygalinc no need; API shape is in the v3 spike (which I need to get back to - so many plates spinning): https://github.com/DapperLib/Dapper/pull/1983

**Comment by brianjlacy** [comment]
Interested to know whether this is still being looked at. Thanks!
